test/unit/core/ds_line_socket_test.dart

```diff
@@ -0,0 +1,156 @@
+
+        await lineSocket.requestAll();
+
+        await Future.delayed(const Duration(milliseconds: 100));
+
+        expect(events.every((event) => event.type == DsDataType.bool && event.value == 1), true);
+      });
+
+      test('send', () async {
+        lineSocketSubscription = lineSocket.stream.listen((event) {
+          return;
+        });
+        final clientSocket = await socketServer.first;
+        await Future.delayed(const Duration(milliseconds: 100));
+        
+        final sendableData = {
+          10: utf8.encode('test1'), 
+          11: utf8.encode('test2'), 
+          12: utf8.encode('test3'), 
+          13: utf8.encode('test4'),
+          14: utf8.encode('test4'),
+          15: utf8.encode('test4'),
+        };
+          
+        String received = '';
+        clientSocket.listen((event) {      
+          received += utf8.decode(event);
+        });
+        
+        String sending = '';
+        for (final event in sendableData.values) {
+          sending += utf8.decode(event);
+          await lineSocket.send(event);
+        }
+        await Future.delayed(const Duration(milliseconds: 1500));
+        print('received: $received');
+        expect(received, sending);
+        received = '';
+        sending = '';
+
+        for (final entry in sendableData.entries) {
+          sending += utf8.decode(entry.value);
+          print('sending: ${utf8.decode(entry.value)}');
+          final result = await lineSocket.send(entry.value);
+          print('result: ${result}');
+          expect(result.hasData, true);
+          await Future.delayed(Duration(milliseconds: entry.key));
+        }
+        await Future.delayed(const Duration(milliseconds: 1000));
+        print('received: $received');
+        expect(received, sending);
+      });
+
+      test('close', () async {
+        lineSocketSubscription = lineSocket.stream.listen((event) {
+          log(true, 'received: ', utf8.decode(event));
+        });
+
+        await Future.delayed(const Duration(milliseconds: 1000));
+        
+        expect(lineSocket.isConnected, true);
+
+        await lineSocket.close();
+
+        /// isConnected = false,
+        expect(lineSocket.isConnected, false);
+        /// exit from _listenSocket()
+        expect(lineSocket.isActive, false);
+      });
+
+      test('stream', () async {
+        final receivedData = <String>[];
+        lineSocketSubscription = lineSocket.stream.listen((event) { receivedData.add(utf8.decode(event)); });
+        final clientSocket = await socketServer.first;
+        
+        await Future.delayed(const Duration(milliseconds: 100));
+        
+        const sendRepetitions = 3;
+        final testMessage = 'test';
+
+        for(var i = 0; i < sendRepetitions; i++) {
+          clientSocket.write(testMessage);
+          await Future.delayed(const Duration(milliseconds: 100));
+        }
+
+        final targetData = List.generate(sendRepetitions, (index) => testMessage);
```

Comment:
remove it

Reply:
Done:
Done:
https://github.com/a-givertzman/hmi_networking/pull/3/files#diff-f7ffc4c31bd14c93001712cc4b0d48993abcecf1fbf7ddfcfb61c55b2eb9ee51